lib/src/components/NxAccordion/stateful/__tests__/NxStatefulAccordion.test.tsx

```diff
@@ -4,46 +4,250 @@
+        )
+      });
+
+      expect(container.querySelector('SUMMARY')).toHaveAttribute('aria-controls', 'foo');
+    });
   });
+
+  describe('accordion header click', function() {
+    describe('when the accordion is currently closed', function() {
+      it('calls onToggle', async function() {
+        const user = userEvent.setup(),
+            onToggle = jest.fn(),
+            el = quickRender({
+              onToggle,
+              children: (
+                <NxAccordion.Header>
+                  <NxAccordion.Title>Foo</NxAccordion.Title>
+                </NxAccordion.Header>
+              )});
+        const header = el.getByRole('button')!;
+        expect(onToggle).not.toHaveBeenCalled();
+
+        await user.click(header);
+
+        expect(onToggle).toHaveBeenCalled();
+      });
+
+      it('toggles the accordion open prop', async function() {
+        const user = userEvent.setup(),
+            onToggle = jest.fn(),
+            el = quickRender({
+              onToggle,
+              children: (
+                <NxAccordion.Header>
+                  <NxAccordion.Title>Foo</NxAccordion.Title>
+                </NxAccordion.Header>
+              )});
+        const header = el.getByRole('button')!;
+        expect(el.getByRole('group')).not.toHaveAttribute('open');
+
+        await user.click(header);
+
+        expect(el.getByRole('group')).toHaveAttribute('open');
+      });
+    });
+
+    describe('when the accordion is currently open', function() {
+      it('onToggle is called', async function() {
+        const user = userEvent.setup();
+        const onToggle = jest.fn();
+
+        const { container } = quickRender({
+          onToggle,
+          defaultOpen: true,
+          children: (
+            <NxAccordion.Header>
+              <span>Foo</span>
+            </NxAccordion.Header>
+          )});
+
+        const header = container.querySelector<HTMLElement>('summary')!;
+
+        expect(onToggle).not.toHaveBeenCalled();
+
+        await user.click(header);
+
+        expect(onToggle).toHaveBeenCalledTimes(1);
+
+        await user.click(container);
+
+        expect(onToggle).toHaveBeenCalledTimes(1);
+      });
+
+      it('toggles the accordion open prop', async function() {
+        const user = userEvent.setup();
+        const onToggle = jest.fn();
+
+        const el = quickRender({
+          onToggle,
+          defaultOpen: true,
+          children: (
+            <NxAccordion.Header>
+              <span>Foo</span>
+            </NxAccordion.Header>
+          )});
+
+        const header = el.getByRole('button')!;
+
+        expect(el.getByRole('group')).toHaveAttribute('open');
+
+        await user.click(header);
+
+        expect(el.getByRole('group')).not.toHaveAttribute('open');
+      });
+    });
+
+    describe('when a non-button element in the header with its own onClick handler is clicked', function() {
+      it('changes the open state and fires onToggle', async function() {
```

Comment:
I don't see assertions in this test about the open state

Reply:
> I don't see assertions in this test about the open state

Same with the test below

Reply:
> I don't see assertions in this test about the open state

Added in here: [7188b0a](https://github.com/sonatype/sonatype-react-shared-components/pull/879/commits/7188b0a816cd9498a090f26065b4f3ac8a75a0c1)